### ISSUE TITLE
ra-data-graphql-simple: Fields dropped with the same type on query

### ISSUE THREAD

**Issue by lnxmad** [marmelab/react-admin#3401]
**What you were expecting:**
All fields are queried on during a GET_ONE

**What happened instead:**
Queries that have the same type defined multiple times are being dropped. It seems that it will ignore any additional fields of the same type if it's already been included.

i.e 

```graphql
{
        Users {
            profile {
                ...
            }
            friends {
                profile {
                    ...
                }
            }
        }
    }
```

The `profile` field is the same type between Users and friends, however, `profile` under friends is left out of the query.

**Steps to reproduce:**
Build schema to include the same type multiple times in a single query

**Related code:**
#3093 This looks to be related.

**Environment**

* React-admin version: 2.9.3
* Last version that did not exhibit the issue (if applicable):2.8.6
* React version: 16.4.1
* Browser: Chrome

**Comment by fzaninotto** [comment]
Can you please provide a CodeSandbox to reproduce the issue?

**Comment by lnxmad** [comment]
This is tough to reproduce without a working server and a schema that is modeled like above. Is there an existing react admin codesandbox with graphql data provider?

**Comment by Kmaschta** [comment]
The demo example runs with Graphql if you set the environment variable  `REACT_APP_DATA_PROVIDER=graphql` feel free to fork it and play with it.

**Comment by lnxmad** [comment]
Is this working?

https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/demo

Getting a spinner.

**Comment by Kmaschta** [comment]
Nope, I think you have an error with your dataProvider because I get the following message in the console:

 ```
TypeError: Failed to fetch
```

**Comment by lnxmad** [comment]
I didn't touch this code. I'm looking for a starting point to recreate.

**Comment by fzaninotto** [comment]
Fixed by #3900
